### PR TITLE
recent search tabs feature enhancement

### PR DESCRIPTION
**Pull request by josh-susinth**
Save recently searched place [#10](https://github.com/OlivierJM/powercut/pull/url)

- will save three recent searches
- enabled cookies to persist the recent searches for 7 days
- Display them under the search field
- Display them in horizontal stack
- Each of these items should be clickable and once clicked it should trigger the find schedule function.
- They should only show before the user has searched.
- [work sample](https://drive.google.com/file/d/1ozK9G4FXnEljzmrdmwqQPUtLMnkmwsxB/view?usp=sharing)

### CODE REVIEW

**Comment by josh-susinth** [comment]
Hi @OlivierJM , I created another PR. Previous PR is now closed as I messed up with yarn.lock.

**Comment by josh-susinth** [comment]
> > Hi @OlivierJM , I created another PR. Previous PR is now closed as I messed up with yarn.lock.
> 
> Thanks @josh-susinth I just had not enough time to review it but I will take a look at this in the coming hours. Thanks a lot for this great work.

sure @OlivierJM looking forward to it. Also to note, this is my first OS contribution so pretty novice with the development environment and checks :)

File: src/components/RecentSearch/RecentSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Button } from '@mantine/core';
+import { IconX } from '@tabler/icons-react';
+
+const RecentSearchTab = ({ onTabSelect, onTabDelete, place }:
```
[the remark this answers is not in the document]
Comment (josh-susinth) [reply src/components/RecentSearch/RecentSearch.tsx:4]: Sure, will do that

File: src/components/Schedule/Finder.tsx [hunk 2]
```diff
@@ -66,6 +72,41 @@ const Finder = () => {
     setCurrentProvince(provArea[0]);
   }, [area]);
 
+  const onOptionSelect = async (place: string) => {
```
[the remark this answers is not in the document]
Comment (josh-susinth) [reply src/components/Schedule/Finder.tsx:75]: Ok. Will remove it

File: src/components/Schedule/Finder.tsx [hunk 3]
```diff
@@ -95,17 +136,46 @@ const Finder = () => {
...
+        justify="flex-start"
+        align="center"
+        direction="row"
+        wrap="wrap"
+        style={{
+          marginBottom: '10px',
+        }}
+      >
+        {
+          recentSearches && recentSearches.split('|').map((place) =>
```
[the remark this answers is not in the document]
Comment (josh-susinth) [reply src/components/Schedule/Finder.tsx:151]: Yeah sure

**Comment by josh-susinth** [comment]
> > > > Hi @OlivierJM , I created another PR. Previous PR is now closed as I messed up with yarn.lock.
> > > 
> > > 
> > > Thanks @josh-susinth I just had not enough time to review it but I will take a look at this in the coming hours. Thanks a lot for this great work.
> > 
> > sure @OlivierJM looking forward to it. Also to note, this is my first OS contribution so pretty novice with the development environment and checks :)
> 
> You did great, just minor things I added comments above and thanks a lot 🚀 🙌🏽

Thank you and will make the changes soon

File: src/components/RecentSearch/RecentSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    { onTabSelect: Function, onTabDelete: Function, place: string }) => (
+        <Button
```
[the remark this answers is not in the document]
Comment (josh-susinth) [reply src/components/RecentSearch/RecentSearch.tsx:6]: Yeah , differentiating the button from main will be good. But centering the horizontal would make the buttons go to next row if the city name is of medium to large length. Currently , it only happens for large length city names. What do you think?

File: src/hooks/useCookie.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+
+function getCookieObject(key: string): string {
```
Comment (Mwimwii) [inline src/hooks/useCookie.ts:3]: change to just `getCookie` since you're returning a string

File: src/hooks/useCookie.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    function setCookie(value: any, daysUntilExpiry?: number): any {
+        if (value !== undefined) {
+            setStoredCookie(value);
+
+            let cookieString = `${key}=${value}`;
+            if (daysUntilExpiry) {
+                const date = new Date();
+                date.setDate(date.getDate() + (daysUntilExpiry));
+                cookieString = `${cookieString}; expires=${date}`;
```
Comment (Mwimwii) [inline src/hooks/useCookie.ts:29]: I think you may require to use `date.toUTCString()` to store GMT time for the expiry

**Review by Mwimwii — CHANGES_REQUESTED** [review]

**Comment by OlivierJM** [comment]
@josh-susinth let us know if you need any help

File: src/hooks/useCookie.ts (hunk 6 above)
Comment (josh-susinth) [reply src/hooks/useCookie.ts:29]: image

its already storing UTC time only

File: src/components/RecentSearch/RecentSearch.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (josh-susinth) [reply src/components/RecentSearch/RecentSearch.tsx]: image
image

is this okay?

[the remark this answers is not in the document]
Comment (OlivierJM) [reply src/components/RecentSearch/RecentSearch.tsx]: This looks okay to me @Mwimwii what do you think ?

[the remark this answers is not in the document]
Comment (Mwimwii) [reply src/components/RecentSearch/RecentSearch.tsx]: Dude, talk about perfection! @josh-susinth 👏👏👏👏

**Comment by josh-susinth** [comment]
> @josh-susinth let us know if you need any help

hey @OlivierJM & @Mwimwii . development is completed and resolved both of your comments too.  Since I'm new to writing unit tests, it would take some time for me to finish writing test cases for this. Though I've written a very basic test case, looks like writing test cases for this feature is a bit more tedious than I expected. I will try to push them ASAP.

**Comment by OlivierJM** [comment]
> > @josh-susinth let us know if you need any help
> 
> hey @OlivierJM & @Mwimwii . development is completed and resolved both of your comments too. Since I'm new to writing unit tests, it would take some time for me to finish writing test cases for this. Though I've written a very basic test case, looks like writing test cases for this feature is a bit more tedious than I expected. I will try to push them ASAP.
> 
> would be helpful if anyone of you can help me with this


@josh-susinth I have added more tests but it is on this branch https://github.com/OlivierJM/powercut/tree/recentFeature can you please sync that branch with yours and push back to this PR then we can merge it. 🙏🏽

**Comment by OlivierJM** [comment]
Or you can merge this https://github.com/josh-susinth/powercut/pull/1/files into your branch and it will reflect here.

**Comment by josh-susinth** [comment]
> Or you can merge this https://github.com/josh-susinth/powercut/pull/1/files into your branch and it will reflect here.

@OlivierJM merged your changes

**Review by OlivierJM — APPROVED** [review]

**Pull request merged**